### ISSUE TITLE
Kafka integration tests fail with plain docker without docker-machine

### ISSUE THREAD

**Issue by yuppie-flu** [druid-io/druid#5159]
I'm trying to run integration tests with Druid inside docker on Linux.
I set `DOCKER_IP` variable to 127.0.0.1
However two integration tests fail:
1. ITKafkaTest.testKafka
2. ITKafkaIndexingServiceTest.testKafka

According to logs, indexing_task for ITKafkaTest failed with the following error
```
2017-12-13T13:30:04,839 ERROR [task-runner-0-priority-0] io.druid.indexing.overlord.ThreadPoolTaskRunner - Exception while running task[RealtimeIndexTask{id=index_realtime_kafka_test_0_2017-
12-13T13:29:15.949Z_jefhkjpp, type=index_realtime, dataSource=kafka_test}]
org.I0Itec.zkclient.exception.ZkTimeoutException: Unable to connect to zookeeper server '127.0.0.1:2181' with timeout of 15000 ms
```
How to run integration tests on linux? Current documentation here: https://github.com/druid-io/druid/tree/master/integration-tests covers only MacOS case.

**Comment by jihoonson** [comment]
@yuppie-flu I guess the IP address is wrong. Would you check it again? If you're using Ubuntu, localhost might be 127.0.1.1. Also, please try ```export DOCKER_IP=$(docker-machine ip integration)```.

**Comment by yuppie-flu** [comment]
@jihoonson I'm using Archlinux and pretty sure that localhost is 127.0.0.1
```
[kozlov@glenovo ~]$ cat /etc/hosts | grep localhost
127.0.0.1	localhost.localdomain	localhost
```
I did `export DOCKER_IP=127.0.0.1`. I can not execute the above command, cause I do not have `docker-machine` command. This tool is only for MacOS and Windows afaik.

**Comment by jihoonson** [comment]
@yuppie-flu then, container's IP address is probably ```172.17.0.1```. Would you check it? If the integration test fails, you can see the containers are still running instead of being stopped on failures. Then, you can get the IP address via several ways like ```docker inspect```.

BTW, docker machine is also available for linux. Please check out https://docs.docker.com/machine/install-machine/.

**Comment by yuppie-flu** [comment]
After deeper investigation I found that current Kafka integration tests setup (`ITKafkaTest` and `ITKafkaIndexingServiceTest`) does not allow to use plain docker without `docker-machine` or separate remote host for running docker containers with druid cluster under tests. It's because these tests assume that druid services inside docker containers can access Kafka broker nodes via ip of docker-machine's vm.
See `integration-tests/docker/Dockerfile`:
```
# unless advertised.host.name is set to docker host ip, publishing data fails
# run this last to avoid rebuilding the image every time the ip changes
ADD docker_ip docker_ip
RUN perl -pi -e "s/#advertised.listeners=.*/advertised.listeners=PLAINTEXT:\/\/$(cat docker_ip):9092/" /usr/local/kafka/config/server.properties
```
In case of plain docker it does not work. You should use something like `$(resolveip -s $HOSTNAME)` instead.

On the other side, `docker-machine` also does not work for me due to this bug:
https://github.com/docker/machine/issues/4349

**Comment by jihoonson** [comment]
@yuppie-flu thank you for the investigation! Would you please submit a patch to fix this issue?

**Comment by yuppie-flu** [comment]
@jihoonson I'm working on it, but it will take some time...

**Comment by jihoonson** [comment]
@yuppie-flu great! Please take your time.

**Comment by yuppie-flu** [comment]
Fixed in #5314
